flight/drone/drone.py

```diff
@@ -0,0 +1,186 @@
+from dronekit import Vehicle
+import logging
+import time
+import threading
+from math import radians, sin, cos
+from pymavlink import mavutil
+
+from optical_flow_attribute import OpticalFlow
+from .. import constants as c
+from ..utils.timer import Timer
+
+class Drone(Vehicle):
+    """Interface to drone and its sensors.
+
+    Attributes
+    ----------
+    _id : int
+        A unique identifier for this drone.
+    """
+
+    def __init__(self, *args):
+        super(Drone, self).__init__(*args)
+        self._id = None
+        self._logger = logging.getLogger(__name__)
+
+        self._optical_flow = OpticalFlow()
+
+        # Allow us to listen for optical flow dat
+        @self.on_message('OPTICAL_FLOW')
+        def listener(self, name, message):
+            """
+            The listener is called for messages that contain the string specified
+            in the decorator,passing the vehicle, message name, and the message.
+            """
+            self._optical_flow.time_usec = message.time_usec
+            self._optical_flow.sensor_id = message.sensor_id
+            self._optical_flow.flow_x = message.flow_x
+            self._optical_flow.flow_y = message.flow_y
+            self._optical_flow.flow_comp_m_x = message.quality
+            self._optical_flow.flow_comp_m_y = message.flow_comp_m_y
+            self._optical_flow.quality = message.quality
+            self._optical_flow.ground_distance = message.ground_distance
+
+            # Notify all observers of new message (with new value)
+            #   Note that argument `cache=False` by default so listeners
+            #   are updaed with every new message
+            self.notify_attribute_listeners('optical_flow', self._optical_flow)
+
+    @property
+    def optical_flow(self):
+        """Get data from the optical flow sensor.
+
+        Notes
+        -----
+        See optical_flow_attribute.py for what kind of data you can get.
+        """
+        return self._optical_flow
+
+    @property
+    def id(self):
+        """Get the drone's id"""
+        return self._id
+
+    @id.setter
+    def id(self, identifier):
+        """Set the drone's id."""
+        self._id = identifier
+
+    def set_attitude(self, roll, pitch, yaw, thrust):
+        """Set the drones attitude.
+
+        Parameters
+        ----------
+        roll : double
```

Comment:
Technically Python has only a float type rather than double, but that's probably nitpicking.

Reply:
Right. I tend to forget that. I will change it.